src/Countries/Spain.php

```diff
@@ -0,0 +1,388 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class Spain extends Country
+{
+    protected function __construct(
+        protected ?string $region = null,
+    ) {
+    }
+
+    public function countryCode(): string
+    {
+        return 'es';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge(
+            [
+                'Año Nuevo' => '01-01',
+                'Epifanía del Señor' => '01-06',
+                'Día del Trabajador' => '05-01',
+                'Asunción de la Virgen' => '08-15',
+                'Fiesta Nacional de España' => '10-12',
+                'Todos los Santos' => '11-01',
+                'Día de la Constitución Española' => '12-06',
+                'Inmaculada Concepción' => '12-08', // 2024?
+                'Navidad' => '12-25'
+            ],
+            $this->variableHolidays($year),
+            $this->regionHolidays($year),
+        );
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easter = $this->easter($year);
+
+        return [
+            'Viernes Santo' => $easter->subDays(2),
+        ];
+    }
+
+    /** @return array<string, string> */
+    protected function regionHolidays(int $year): array
+    {
```

Comment:
```php
$holidays = Holidays::for(Spain::make(year: 2020, region: 'es-an'))->get(); 
```

In this case I think we should throw an exception, as we won't be returning the region based holidays.

Reply:
I see, and implement some direct way to also return national holidays, which are still valid, when there is no regional information for a specific year?

Reply:
You can use InvalidYear::rang()